### PR TITLE
Add LDFLAGS params to enable testbuilds.

### PR DESCRIPTION
**Pull request by dschreij**
Some libraries (such as libvorbis or various sdl2 modules) try to compile and run small test programs at build time to see if a dependency is correctly installed. They often fail on TravisCI, even when the dependency library is there, and adding this parameter fixes this (thanks to @jakirkham).

See https://github.com/conda-forge/staged-recipes/pull/2128 for an example

### CODE REVIEW

File: recipe/activate.sh [hunk 1]
```diff
@@ -14,6 +14,7 @@ then
     export LDFLAGS="${LDFLAGS} -headerpad_max_install_names"
     export LDFLAGS="${LDFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
     export LDFLAGS="${LDFLAGS} -lc++"
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,${PREFIX}/lib"
```
Comment (jakirkham) [inline recipe/activate.sh]: Could we please add this for Linux too? Would change to `-Wl,-rpath=$PREFIX/lib`.

**Comment by jakirkham** [comment]
Please bump the minor version number, @dschreij .

**Comment by jakirkham** [comment]
Thanks for putting this together. 👍

**Comment by isuruf** [comment]
I proposed #11 which is similar to this, but reviewers didn't like putting anything more to toolchain.

**Comment by dschreij** [comment]
Alright, I was requested to make this PR by one of the conda crew (who also commented on your PR I see now). I'd say we've done our duty ;)

**Comment by scopatz** [comment]
Yeah @jakirkham and I requested this.  I don't have merge rights here, though.

**Comment by jakirkham** [comment]
There's a failure likely due to the fact this feedstock has not been re-rendered in a long time. Adding PR ( https://github.com/conda-forge/toolchain-feedstock/pull/19 ) to fix that.

**Comment by isuruf** [comment]
Not needed anymore due to #23

File: recipe/activate.sh [hunk 2]
```diff
@@ -32,6 +33,7 @@ then
     #
     export CXXFLAGS="${CXXFLAGS} -DBOOST_MATH_DISABLE_FLOAT128"
     export LDFLAGS="${LDFLAGS}"
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,$PREFIX/lib"
```
Comment (jakirkham) [inline recipe/activate.sh]: On Linux, this need to be `=` instead of `,`. IOW `-Wl,-rpath=$PREFIX/lib`.

Comment (isuruf) [reply recipe/activate.sh]: Both are correct in linux right?

**Comment by jakirkham** [comment]
Thanks for resurrecting this, @isuruf. Was just thinking about putting the same thing together and totally forgot we already had it here. Added one minor comment above.

File: recipe/activate.sh (hunk 2 above)
Comment (jakirkham) [reply recipe/activate.sh]: Hmm...I don't think so, but it has been a while since I checked.

Comment (isuruf) [reply recipe/activate.sh]: I've been using it on many recipes without any problems.

Comment (jakirkham) [reply recipe/activate.sh]: Do you have an example?

Comment (isuruf) [reply recipe/activate.sh]: https://github.com/conda-forge/gmp-feedstock/blob/2d45981ccd505a8049bedd74e91ea58ac16058b3/recipe/run_test.sh

Comment (jakirkham) [reply recipe/activate.sh]: Not seeing any warnings in the CI log. So guess it works fine.

**Pull request merged**